### PR TITLE
Update Contact Merge page DRS query

### PR DESCRIPTION
**Pull request by psadie-sfdo**
[W-8522759](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008kXYEIA2/view)

This change allows Duplicate Record Set records that are created via a Duplicate Job to appear on the NPSP Contact Merge page. Duplicate Record Set records created via a Duplicate Job do not have their Duplicate Rule lookup populated.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/CON_ContactMerge_CTRL.cls [hunk 1]
```diff
@@ -1105,27 +1105,53 @@ public with sharing class CON_ContactMerge_CTRL {
...
-            .withSelectFields(driFieldNames)                                  
-            .build();
-        String subQueryStr = '('+subQuery+')';
-        drsFieldNames.add(subQueryStr);
-        String setOrderBy = 'LastModifiedDate DESC ';
-        String setWhereClause = 'RecordCount > 1 AND DuplicateRule.SObjectType = \'Contact\'';
         //build the SOQL query and execute
-        String queryStr =  new UTIL_Query()
+        List<String> forDuplicateRecordSet = getDuplicateRuleSetQuerySelectClause();
+        String forDuplicateRecordSetsRelatedToAtleastOneContact = getDuplicateRuleSetQueryWhereClause();
```
Comment (voduyemi) [inline src/classes/CON_ContactMerge_CTRL.cls:1110]:
```suggestion
        String forDuplicateRecordSetsRelatedToAtLeastOneContact = getDuplicateRuleSetQueryWhereClause();
```

File: src/classes/CON_ContactMerge_CTRL.cls [hunk 2]
```diff
@@ -1105,27 +1105,53 @@ public with sharing class CON_ContactMerge_CTRL {
...
+
+        String duplicateRecordSetsQuery =  new UTIL_Query()
             .withFrom(DuplicateRecordSet.sObjectType)
-            .withSelectFields(drsFieldNames)
-            .withWhere(setWhereClause)
-            .withOrderBy(setOrderBy)
+            .withSelectFields(forDuplicateRecordSet)
+            .withWhere(forDuplicateRecordSetsRelatedToAtleastOneContact)
```
Comment (voduyemi) [inline src/classes/CON_ContactMerge_CTRL.cls:1115]:
```suggestion
            .withWhere(forDuplicateRecordSetsRelatedToAtLeastOneContact)
```

**Review by voduyemi — APPROVED** [review]
This looks good! @psadie-sfdo. Just had a naming suggestion.

**Review by sokosaurus — APPROVED** [review]
Looks good to me! Thanks for the opportunity to review!

**Pull request merged**